### PR TITLE
Add top level README.md

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
Just adds a short description as well as links to the build instructions and the wiki

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# GNAT2GOTO
+
+A [Ada](https://en.wikipedia.org/wiki/Ada_(programming_language)) frontend for [CBMC](https://github.com/diffblue/cbmc). Named thus because it is implemented as a backend for the [GNAT Ada Compiler](https://www.gnu.org/software/gnat/) (see also: [AdaCore](https://www.adacore.com/)), that produces a GOTO program (CBMCs internal program representation) in JSON format.
```
[the remark this answers is not in the document]
Comment (hannes-steffenhagen-diffblue) [reply README.md]: Oxford dictionary says no: https://en.oxforddictionaries.com/punctuation/apostrophe (see "Apostrophes and plural forms")

File: README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+For build instructions, see [gnat2goto/README.md](gnat2goto/README.md).
```
[the remark this answers is not in the document]
Comment (hannes-steffenhagen-diffblue) [reply README.md]: Considered it, genuinely not sure which would be better. Some projects have build instructions in their top level, some don't :man_shrugging:

I certainly don't mind either way if you prefer this

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (xbauch) [reply README.md]: And how many CBMCs are we talking about?

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisr-diffblue) [reply README.md]: True. Maybe then keep your new top-level README.md, but rename the gnat2goto/README.md to COMPILING.md?

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisr-diffblue) [reply README.md]: Simple cheat answer (my grammar is terrible): "the internal program representation of CBMC"

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hannes-steffenhagen-diffblue) [reply README.md]: I was rewording it anyway, so might as well do it now

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hannes-steffenhagen-diffblue) [reply README.md]: Rewrite it is! Either way doesn't read great with or without the '

[the remark this answers is not in the document]
Comment (NlightNFotis) [reply README.md]: *nit*: A`n` Ada frontend...

Comment (peterschrammel) [inline README.md]: A Ada -> An Ada

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (peterschrammel) [reply README.md]: I think keeping the README separate from COMPILING and at the top level is good. In any case you can link them.

**Comment by hannes-steffenhagen-diffblue** [comment]
I've updated the spelling and and wording, and ended up rewording gnat2goto/README.md (now gnat2goto/COMPILING.md)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# GNAT2GOTO
+An [Ada](https://en.wikipedia.org/wiki/Ada_(programming_language)) frontend for
+[CBMC](https://github.com/diffblue/cbmc). Named thus because it is implemented
+as a backend for the [GNAT Ada Compiler](https://www.gnu.org/software/gnat/)
+(see also: [AdaCore](https://www.adacore.com/)), that produces a GOTO program
+(the internal program representation of CBMC) in JSON format.
+
+For instructions on building the project, see [gnat2goto/BUILDING.md](gnat2goto/BUILDING.md)
```
Comment (chrisr-diffblue) [inline README.md]: COMPILING.md?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+For an overview of the project, see the [Wiki](https://github.com/diffblue/gnat2goto/wiki) (work in progress)
```
Comment (chrisr-diffblue) [inline README.md]: Drop the 'work in progress' - it's *always* going to be a work in progress :-)

Comment (hannes-steffenhagen-diffblue) [reply README.md]: Ah, misplaced optimism on my part

File: README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# GNAT2GOTO
+An [Ada](https://en.wikipedia.org/wiki/Ada_(programming_language)) frontend for
+[CBMC](https://github.com/diffblue/cbmc). Named thus because it is implemented
+as a backend for the [GNAT Ada Compiler](https://www.gnu.org/software/gnat/)
+(see also: [AdaCore](https://www.adacore.com/)), that produces a GOTO program
+(the internal program representation of CBMC) in JSON format.
+
+For instructions on building the project, see [gnat2goto/COMPILING.md](gnat2goto/BUILDING.md)
```
Comment (chrisr-diffblue) [inline README.md]: *almost* :-) BUILDING.md -> COMPILING.md :-)

File: gnat2goto/COMPILING.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Build Dependencies
+
+**To build GnatColl GPL 2016 and gnat2goto you must use a gnat GPL 2016 compiler and gprbuild (see the GnatColl user documentation)**
+
+We need the 2016 versions of the GNAT Ada compiler, the gprbuild system and the
+GNATColl utility library to build GNAT2GOTO.
+
+Note: you will *not* be able to build using the debian/ubuntu gnat and gprbuild.
```
Comment (chrisr-diffblue) [inline gnat2goto/COMPILING.md]: Is it worth expanding that note a little bit - "gnat2goto needs the specific versions of GNAT listed in this file. Debian/Ubuntu packaged builds of gnat and gprbuild do not match this version."

File: README.md (hunk 5 above)
Comment (hannes-steffenhagen-diffblue) [reply README.md]: argh

File: gnat2goto/COMPILING.md (hunk 6 above)
Comment (hannes-steffenhagen-diffblue) [reply gnat2goto/COMPILING.md]: I'm actually not sure how specific it needs to be. It doesn't work with the Ubuntu version, but the it works fine with "the" version presented on AdaCore as the 2016 version.

**Comment by hannes-steffenhagen-diffblue** [comment]
The rewording of the now `BUILDING.md` was basically merging the Debian and Mac OSX instructions at they were almost identical except for a few details, which are instead pointed out separately now

File: gnat2goto/COMPILING.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+For **Debian**, install the following debian packages:
+  - build-essential
+
+For **Mac OSX**, you will need the XCode Command Line tools, which require XCode itself and then a 
+```
+xcode-select --install
+```
+to make them available.
+
+Go to [GNAT Community Download page at AdaCore](https://www.adacore.com/download/more),
```
Comment (chrisr-diffblue) [inline gnat2goto/COMPILING.md]: This is an action that is common to both OSX and Debian, but the way this documentation is structured makes it look like you only need to do this for Mac OS X.  I'd suggest moving this before the Debian/OSX sections, maybe? Or giving it a heading that makes it clearer that this is needed for both platforms?

Comment (chrisr-diffblue) [reply gnat2goto/COMPILING.md]: Actually, it has to be after the Debian/OSX sections, of course. So maybe give this section a "Installing GNAT and GNATCOLL" title?

Comment (hannes-steffenhagen-diffblue) [reply gnat2goto/COMPILING.md]: Has heading and separator line now

**Comment by hannes-steffenhagen-diffblue** [comment]
CI fails because of unrelated issue about changes on the CBMC side (we should really start documenting what version of CBMC we last ran the tests with once the json goto branch is merged)

**Comment by hannes-steffenhagen-diffblue** [comment]
I've added https://github.com/diffblue/gnat2goto/issues/63 because that's something we really should address actually

**Pull request merged**